pages/contacts/[id]/index.js

```diff
@@ -32,30 +33,22 @@ const StyledInfo = styled.p`
   padding: 1rem;
   border-radius: 8px;
 `;
-
+//----------------------------------------------- FUNCTION------------HERE
 export default function ContactDetail() {
   const router = useRouter();
-  const { dynamicId } = router.query;
-
-  const contact = contacts.find((contact) => contact.id === dynamicId);
-
-  if (!contact) {
-    return <p>Kontakte nicht gefunden</p>;
-  }
+  const { query } = router;
+  const { id } = query;
+  const { data: contact, isLoading, error } = useSWR(`/api/contacts/${id}`);
 
-  const { name, photo, phone, note } = contact;
+  if (isLoading) return <h2>Loading...</h2>;
+  if (error) return <h2>Error...</h2>;
 
   return (
     <>
       <StyledHeader>
         <StyledHeaderContainer>
-          <StyledH3>{name}</StyledH3>
-          <StyledImage
-            src={photo}
-            width={34}
-            height={34}
-            alt={`Photo of ${name}`}
-          />
+          <StyledH3>{contact.name}</StyledH3>
```

Comment:
I think it should be better to decide for name that is more descriptive. `StyledHeading` for example? 🤔